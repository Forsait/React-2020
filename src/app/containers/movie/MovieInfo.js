import React from 'react';
import { connect } from 'react-redux';

import { getMovieInfo } from '../../actions/movie-info';

import styles from './MovieInfo.module.scss';
import Header from '../../components/common/Header';
import Westside from '../../components/common/Westside';
import { getYear } from '../../utils/date';

import searchIcon from 'Assets/icon-search.svg';

export class MovieInfo extends React.Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.getMovieInfo(353081);
  }

  render() {
    let genreView = null;
    if (this.props.movie.genres) {
      genreView = <div>Film by <span className={styles.som}>{this.props.movie.genres[0]}</span> genre</div>;
    }

    return <>
      <div className={styles.container}>
        <Header>
          <a href="#" className={styles.search_link}>
            <img src={searchIcon} />
          </a>
        </Header>

        <div className={styles.movie}>
          <img src={this.props.movie.poster_path} className={styles.poster}/>
          <div className={styles.info}>
            <div className={styles.title_box}>
              <h1 className={styles.title}>{this.props.movie.title}</h1>
              <span className={styles.rating}>{this.props.movie.vote_average}</span>
            </div>
            <div className={styles.subtitle}>Oscar winning movie</div>
            <div className={styles.maus}>
              <div className={styles.t4}>
                <span className={styles.t4_1}>{ getYear(this.props.movie.release_date) }</span>
                <span className={styles.t4_2}>year</span>
              </div>
              <div className={styles.t4}>
                <span className={styles.t4_1}>{this.props.movie.runtime || '-'}</span>
                <span className={styles.t4_2}>min</span>
              </div>
            </div>
            <div className={styles.description}>{this.props.movie.overview}</div>
          </div>
        </div>
      </div>
      <Westside className={styles.hot}>
        {genreView}
      </Westside>
    </>
  }
}

function mapStateToProps(state) {
  const { movieInfo } = state.movieInfo;
  return { movie: movieInfo };
}

const mapDispatchToProps = dispatch => {
  return {
    getMovieInfo: id => dispatch(getMovieInfo(id))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(MovieInfo);