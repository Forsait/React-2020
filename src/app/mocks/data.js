export const film = {
  budget: 0,
  genres: ['Action', 'Adventure', 'Drama', 'Science Fiction', 'Thriller'],
  id: 351286,
  overview: 'A volcanic eruption threatens',
  poster_path: 'https://image.tmdb.org/t/p/w500/ln6d5Okr6VK5vfQVobJTiYxeD0l.jpg',
  release_date: '2018-06-01',
  revenue: 0,
  runtime: null,
  tagline: 'Life finds a way',
  title: 'Jurassic World: Fallen Kingdom',
  vote_average: 0,
  vote_count: 28,
};

export const moviesData = {
  data: new Array(6).fill(film),
  limit: 0,
  offset: 0,
  total: 0,
};

export const radioOptions = {
  title: 'Radio title',
  list: [
    {
      key: 'key 1',
      title: 'title 1',
    },
    {
      key: 'key 2',
      title: 'title 2',
    },
  ],
};
