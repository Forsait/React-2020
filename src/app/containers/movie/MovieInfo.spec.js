import React from 'react';
import { shallow } from 'enzyme';

import { MovieInfo } from './MovieInfo';
import { film } from '../../mocks/data.js';

describe('MovieInfo component', () => {

  it('should be render correctly', () => {
    MovieInfo.prototype.componentDidMount = jest.fn();
    const com = shallow(<MovieInfo movie={film} />);
    expect(com).toMatchSnapshot();
  });

});
