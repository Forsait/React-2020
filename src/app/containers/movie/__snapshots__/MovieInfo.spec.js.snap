// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo component should be render correctly 1`] = `
<Fragment>
  <div>
    <Header>
      <a
        href="#"
      >
        <img
          src={
            Object {
              "ReactComponent": "SvgMock",
            }
          }
        />
      </a>
    </Header>
    <div>
      <img
        src="https://image.tmdb.org/t/p/w500/ln6d5Okr6VK5vfQVobJTiYxeD0l.jpg"
      />
      <div>
        <div>
          <h1>
            Jurassic World: Fallen Kingdom
          </h1>
          <span>
            0
          </span>
        </div>
        <div>
          Oscar winning movie
        </div>
        <div>
          <div>
            <span>
              2018
            </span>
            <span>
              year
            </span>
          </div>
          <div>
            <span>
              -
            </span>
            <span>
              min
            </span>
          </div>
        </div>
        <div>
          A volcanic eruption threatens
        </div>
      </div>
    </div>
  </div>
  <Westside>
    <div>
      Film by 
      <span>
        Action
      </span>
       genre
    </div>
  </Westside>
</Fragment>
`;
